report/main.py

```diff
@@ -0,0 +1,476 @@
+import base64
+from datetime import datetime, timedelta
+from os import environ, _Environ
+
+import altair as alt
+from altair.vegalite.v5.api import Chart
+from dotenv import load_dotenv
+import pandas as pd
+from pandas import DataFrame
+from psycopg2 import connect
+from psycopg2.extensions import connection
+
+
+from email.mime.multipart import MIMEMultipart
+from email.mime.application import MIMEApplication
+
+from xhtml2pdf import pisa
+import plotly_express as px
+import boto3
+
+
+def get_db_connection(config_file: _Environ) -> connection:
+    """
+    Returns connection to the database
+
+    Args:
+        config (_Environ): A file containing sensitive values
+
+    Returns:
+        connection: A connection to a Postgres database
+    """
+    try:
+        return connect(
+            database=config_file["DB_NAME"],
+            user=config_file["DB_USER"],
+            password=config_file["DB_PASSWORD"],
+            port=config_file["DB_PORT"],
+            host=config_file["DB_HOST"]
+        )
+    except Exception as err:
+        print("Error connecting to database.")
+        raise err
+
+
+def get_database(conn_postgres: connection) -> DataFrame:
+    """
+    Returns redshift database transaction table as a DataFrame Object
+
+    Args:
+        conn_postgres (connection): A connection to a Postgres database
+
+    Returns:
+        DataFrame:  A pandas DataFrame containing all relevant release data
+    """
+    query = f""
+    df_releases = pd.read_sql_query(query, conn_postgres)
+
+    return df_releases
+
+
+def convert_html_to_pdf(source_html, output_filename):
+    """
+    """
+    # open output file for writing (truncated binary)
+    result_file = open(output_filename, "w+b")
+
+    # convert HTML to PDF
+    pisa_status = pisa.CreatePDF(
+        source_html,                # the HTML to convert
+        dest=result_file)           # file handle to recieve result
+
+    # close output file
+    result_file.close()                 # close output file
+
+    return pisa_status.err
+
+
+def get_new_releases(df_releases: DataFrame) -> int:
+    """
+    Return the number of new releases for the previous day
+
+    Args: 
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        int: An integer relating to the number of new games released
+    """
+    date = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
+
+    return df_releases[df_releases["release_date"] == date].reset_index().shape[0]
+
+
+def get_top_rated_release(df_releases: DataFrame) -> str:
+    """
+    Return the name of new release with the highest overall sentiment score for the previous day
+
+    Args: 
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        str: A string relating to the title of the highest rated new game released
+    """
+    date = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
```

Comment:
104 and 105 -- as mentioned below make into a new function as you've got it repeated